cassandra/components.py

```diff
@@ -1161,3 +1160,390 @@ def run_component(self):
+
+        :param yr:                      Target year
+        :type yr:                       int
+
+        :param min_yr:                  Minimum year allowable
+        :type min_yr:                   int
+
+        :param max_yr:                  Maximum year allowable
+        :type max_yr:                   int
+
+        :return:                        [0] validated start year as a four digit integer
+                                        [1] validated end year as a four digit integer
+
+        """
+
+        # validate that the year can be converted to an integer
+        valid_yr = self.validate_int(yr)
+
+        if (valid_yr < min_yr) or (valid_yr > max_yr):
+            msg = f"{self.__class__} Year '{valid_yr}' is outside of the reasonable bounds [{min_yr} - {max_yr}]."
+            self.log_raise_exception(ValueError, msg)
+
+        else:
+            return valid_yr
+
+    def validate_int(self, value):
+        """Ensure the value can be converted to an integer.
+
+        :param value:                   Target value
+        :type value:                    int, float, str
+
+        :return:                        Validated int of value
+
+        """
+
+        if type(value) is int:
+            return value
+
+        try:
+            return int(value)
+
+        except ValueError:
+            msg = f"{self.__class__} Value '{value}' not able to be converted to an integer as expected."
+            self.log_raise_exception(ValueError, msg)
+
+    def validate_file_exist(self, file_path):
+        """Ensure file exists.
+
+        :param file_path:               Full path with file name an extension to in input file.
+        :type file_path:                str
+
+        :return:                        Validated file path
+
+        """
+
+        if os.path.isfile(file_path):
+            return file_path
+
+        else:
+            msg = f"Input file '{file_path}' does not exist."
+            self.log_raise_exception(FileNotFoundError, msg)
+
+    def rds_to_dict(self):
+        """Read in and convert an RDS file to a Python dictionary.
+
+        :return:                        A Python dictionary of {variable name: value arrays}
+
+        """
+
+        import rpy2.robjects as robjects
+        from rpy2.robjects import pandas2ri
+        pandas2ri.activate()
+
+        # get a wrapper around the readRDS R function
+        read_rds = robjects.r['readRDS']
+
+        # create ListVector object
+        rds_file = self.validate_file_exist(self.params[self.RDS_FILE_FIELD])
+        lvect = read_rds(rds_file)
+
+        # convert the ListVector object to a Python dictionary
+        return dict(zip(lvect.names, map(list, list(lvect))))
+
+    def build_file_dict(self, rds_dict):
+        """Get a list of target files that match the climate variable specified in the configuration.
+
+        :param rds_dict:                A Python dictionary of {variable name: value arrays} derived from the emulator
```

Comment:
I'm not sure I understand what is going on here.  Are you extracting the names of netCDF files from the RDS file?  I don't think it's necessary to go back to the original data to extract this information.  I'm pretty sure we store the global mean temperature series in the fldgen save file.  If not, then it would be easy to do so, since we have to calculate it as part of the training process.  We _definitely_ don't want to be going back to the original data.  First, it's brittle (What if the original data files move?  Or what if you want to take the calculation to a new computing system?), and second it's unnecessarily slow.